internal/error.go

```diff
@@ -222,6 +222,12 @@ type (
 		cause        error
 	}
 
+	// ActivityNotRegisteredError is returned if worker doesn't support activity type.
+	ActivityNotRegisteredError struct {
```

Comment:
Should we just use `ApplicationError` directly instead of creating a new error type.  I think there are more locations in the code where conversions to Error and Failures are happening.  Having `ActivityNotRegisteredError` as a top level might cause issues in other places.  For instance how do you control `IsRetry` behavior for this error class?

Reply:
It's an internal error that does not get exposed to the user, it will get translated to the `ApplicationError` down [the line](https://github.com/temporalio/sdk-go/blob/a39bbe82b2bab1281d24d7f18a93a9813e0e622b/internal/error.go#L889).